paths/paths.go

```diff
@@ -59,3 +59,13 @@ func DefaultTailscaledStateFile() string {
 	}
 	return ""
 }
+
+// MkStateDir ensures that the directory dirPath both exists and has the correct
```

Comment:
elaborate on what the "correct permissions" are?

And maybe what "StateDir" is? (that it's meant for the system daemon's configuration directory containing the machine keys, etc)